x-pack/plugin/migrate/src/main/java/org/elasticsearch/xpack/migrate/task/ReindexDataStreamEnrichedStatus.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.migrate.task;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.core.Tuple;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/*
+ * This class represents information similar to that in ReindexDataStreamStatus, but enriched from other sources besides just the task
+ * itself.
+ */
+public record ReindexDataStreamEnrichedStatus(
+    long persistentTaskStartTime,
+    int totalIndices,
+    int totalIndicesToBeUpgraded,
+    boolean complete,
+    Exception exception,
+    Map<String, Tuple<Long, Long>> inProgress,
+    int pending,
+    List<Tuple<String, Exception>> errors
+) implements Task.Status {
+    public ReindexDataStreamEnrichedStatus {
+        Objects.requireNonNull(inProgress);
+        Objects.requireNonNull(errors);
+    }
+
+    public static final String NAME = "ReindexDataStreamStatus";
```

Comment:
should this be `ReindexDataStreamEnrichedStatus` ?

Reply:
Actually this class has no business implementing Task.Status, so it has no need of a name. I'll fix that.